module_utils/network_lsr/argument_validator.py

```diff
@@ -633,6 +704,177 @@ def _validate_post(self, value, name, result):
         return result
 
 
+class ArgValidatorIPRoutingRules(ArgValidatorDict):
+    def __init__(self, name, family=None, required=False):
+        ArgValidatorDict.__init__(
+            self,
+            name,
+            required,
+            nested=[
+                # a valid priority ranges from 0 to 0xFFFFFFFF, "-1" is also allowed
+                # to reset the priority
+                ArgValidatorNum(
+                    "priority",
+                    default_value=None,
+                    required=True,
+                    val_min=-1,
+                    val_max=0xFFFFFFFF,
+                ),
+                ArgValidatorStr(
+                    "action",
+                    default_value="to-table",
+                    enum_values=["to-table", "blackhole", "prohibit", "unreachable"],
+                ),
+                ArgValidatorRange("dport", val_min=0, val_max=65534),
+                ArgValidatorStr(
+                    "family",
+                    default_value="ipv4",
+                    enum_values=["ipv4", "ipv6"],
+                ),
+                ArgValidatorIP(
+                    "from", family=family, default_value=None, plain_address=False
+                ),
+                ArgValidatorNum("from_len", default_value=0, val_min=0),
+                ArgValidatorNum(
+                    "fwmark", default_value=None, val_min=0, val_max=0xFFFFFFFF
+                ),
+                ArgValidatorNum(
+                    "fwmask", default_value=None, val_min=0, val_max=0xFFFFFFFF
+                ),
+                ArgValidatorStr("iif", default_value=None),
+                ArgValidatorBool("invert", default_value=None),
+                ArgValidatorNum("ipproto", default_value=None, val_min=0, val_max=255),
+                ArgValidatorStr("oif", default_value=None),
+                ArgValidatorRange("sport", val_min=0, val_max=65534),
+                ArgValidatorNum(
+                    "suppress_prefixlength", default_value=None, val_min=-1
+                ),
+                ArgValidatorRouteTable("table"),
+                ArgValidatorIP(
+                    "to", family=family, default_value=None, plain_address=False
+                ),
+                ArgValidatorNum("to_len", default_value=0, val_min=0),
+                ArgValidatorNum("tos", default_value=0, val_max=255),
+                ArgValidatorRange("uid", val_min=0, val_max=0xFFFFFFFF),
+            ],
+            default_value=None,
+        )
+        self.family = family
+
+    def _validate_post(self, value, name, result):
+        result["family"] = (socket.AF_INET, socket.AF_INET6)[result["family"] == "ipv6"]
```

Comment:
`(socket.AF_INET if (result["family"] == "ipv4") else socket.AF_INET6)`

also, it would seem convenient to detect the family based on to/from (if given). And, to/from must agree to have the same address family.

Reply:
Your proposal works, but I think that it more like the personal taste.

For the `suppress_prefixlength` setting, we need to know the IP family, that is the primary reason that I introduced `family` as an attribute.

Reply:
> Your proposal works, but I think that it more like the personal taste.

Where did you see the pattern `[a, b](bool_val)`? While it works, it does not seem common to me (never saw it) and in my opinion, it's not a good pattern.

> For the suppress_prefixlength setting, we need to know the IP family, that is the primary reason that I introduced family as an attribute.

Of course you need to know the family. That means, at least one of `family`, `to`, `from` must be specified. If several of them are specified,then their family must agree.

This does not mean that `family` should be mandatory, if you can deduce it from `from`/`to`. Requiring the user to provide redundant information is cumbersome (for the user).

Reply:
> Where did you see the pattern [a, b](bool_val)? While it works, it does not seem common to me (never saw it) and in my opinion, it's not a good pattern.

i don't like python's `val1 if condition else val2` either. I think it's often hard to read, but this is the ternary operator in Python.

I often instead write
```
   if condition:
        val = val1
   else:
        val = val2
```
but that's not great too (verbose).